### ISSUE TITLE
Parse error in multiline attribute expression

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#35415]
**Godot version:**
master @ d11d7dfe3e15e34ba400e6b73d618abc4b8f5886

**OS/device including version:**
Linux

**Issue description:**
Godot raises parse error on the code below:
```
class X:
	func foo(p):
		var x = (
			p
			.some
			.some
			.some
		)
```
```
Godot Engine v3.2.rc.custom_build.d11d7dfe3 - https://godotengine.org
 
SCRIPT ERROR: GDScript::reload: Parse Error: Expected ')' in expression
   At: res://tests/potential-godot-bugs/multiline-attribute-expression.gd:5.
ERROR: reload: Method/Function Failed, returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:576.
```

**Comment by Calinou** [comment]
I think the only way to chain methods on newlines right now is to add backslashes at the end of every line.

**Comment by Scony** [comment]
Well, this is a bit counter-intuitive since python (which GDScript is motivated by) has no problems with such a code. Also, it's useful for formatting in case of such a long attr/call chains.

**Comment by Calinou** [comment]
The issue is that the `.method()` syntax is used to call a method on the parent class. Therefore, it'd be ambiguous to support chaining methods without a backslash.

**Comment by Scony** [comment]
Actually it doesn't matter. You cannot have an attribute expression starting with a dot (like `.aaa.bbb.ccc`). And hence there is no ambiguity - from pure parsing perspective, the expression starting with dot must be a parent call. And please note that newline should not matter inside of any expression (maybe except long string `"""xxx"""`).

**Comment by Calinou** [comment]
@Scony Right, we can support it for property access but not for method calls. That's pretty inconsistent, so I'm still not sure if we should support that :slightly_frowning_face:

**Comment by Scony** [comment]
@Calinou what do you mean by *but not for method calls*?

**Comment by dalexeev** [comment]
```gdscript
var x = (
    p
    .some
    .some
    .some
)
```

I am not a fan of this style of code, but specifically in this case, the parser is wrong. Since the expression is enclosed in parentheses, indents should be treated as spaces:

```gdscript
var x = (p .some .some .some)
```

Probably the problem is that in the current implementation of the parser, indentation is treated the same way, regardless of context (operator stream or expression).

**Comment by bojidar-bg** [comment]
> Probably the problem is that in the current implementation of the parser, indentation is treated the same way, regardless of context (operator stream or expression).

Actually the problem is simpler: The code in the parser currently does not try parsing newlines interspersed in indexing operations, the way the code related to parsing parts of an expression does. (and, no, `x.y` is not really an operator, since `y` is not a valid expression on its own)

Code not checking for a newline between `x<newline>.y`
https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L1178-L1182

Code checking for a newline between `x +<newline>y`:
https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L255-L260

<details>
<summary>
Code checking for a newline in `x<newline>+ y`:
</summary>

https://github.com/godotengine/godot/blob/1c6d1fcf956b882fc8a14fa595b7f1e9b15e1574/modules/gdscript/gdscript_parser.cpp#L1295-L1300

</details>

Keep in mind that a similar check would be needed for `p.<newline>some`, and maybe `p.some<newline>(`.

Finally, note that the following is valid syntax:
```gdscript
func a():
    var b = c # Important: no parentheses around c<newline>.d()
    .d()
```

It stands for a call to `d()` in the base/super class.

**Comment by JohnCorby** [comment]
cant this pretty easily be fixed with 4.0 since it will use `super.thing` instead of just `.thing` to get superclass methods/fields? when that happens, doing this chaining (with or without parenthesis) wont be ambiguous for the parser, right?

**Comment by KoBeWi** [comment]
Fixed in 4.0
